CMake reorg: cmakelists in each dir

biagas opened the pull request:
### Description

Part of the CMake Re-organization effort.

The goal of this PR isn't to introduce new CMake logic in most cases, but to have CMakeLists.txt live as close to source as possible by adding CMakeLists.txt to subdirectories that previously didn't have one.
Most of the subdirs CMakeLists.txt simply add their source files and include directories to the parent's library, unless there were other steps needed, like adding the code-gen, or creating a configure file.


### Type of change

<!-- Please check one of the boxes below -->

~~* [ ] Bug fix~~
~~* [ ] New feature~~
~~* [ ] Documentation update~~
* [X] Other~~ <!-- please explain with a note below -->
CMake re-organization.

### How Has This Been Tested?

Compiled and ran regression suite on Linux successfully.
Successfully compiled on Windows.

### Reminders:

- Please follow the [style guidelines][1] of this project.
- Please perform a self-review of your code before submitting a PR and asking others to review it.
- Please assign reviewers (see [VisIt's PR procedures][2] for more information).

### Checklist:

<!-- For items in this checklist that do not apply, simply insert two tilde chars, `~~`, just ahead of the left bracket char, `[` at the beginning of a line. Each line ends with two tilde chars to make doing such ~~strikeouts~~ easy. -->

- [ ] I have commented my code where applicable.~~
- [ ] I have updated the release notes.~~
- [ ] I have made corresponding changes to the documentation.~~
- [ ] I have added debugging support to my changes.~~
- [ ] I have added tests that prove my fix is effective or that my feature works.~~
- [ ] I have confirmed new and existing unit tests pass locally with my changes.~~
- [ ] I have added new baselines for any new tests to the repo.~~
- [ ] I have NOT made any changes to [*protocol* or *public interfaces*][3] in an RC branch.~~

[1]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/StyleGuide.html
[2]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/pr_create.html#reviewers
[3]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/RCDevelopment.html#communication-protocols-and-public-apis

biagas committed:
Reorganize common.

Add CMakeLists in each directory, which lists its sources and adds them to visitcommon library target.

biagas committed:
Cleanup common

biagas committed:
Reorg avt/Database

biagas committed:
Reorg avt/DBAtts.

biagas committed:
Reorg avt/Expressions.

biagas committed:
Reorg avt/MIR

biagas committed:
Reorg avt/Pipeline.

biagas committed:
Reorg avt/Plotter.

biagas committed:
Reorg avt/QtVisWindow.

biagas committed:
Reorg avt/Queries.

biagas committed:
Reorg avt/VisWindow.

biagas committed:
Fix link issue with avtquery.

biagas committed:
Revert changes to Volume plot CMakeList.txt

biagas committed:
Merge branch 'develop' into task/biags/cmake_reorg_cmakelists_each_dir

cyrush commented on src/avt/Database/CMakeLists.txt:
```diff
@@ -8,99 +8,55 @@
-
-# Add link directories
-LINK_DIRECTORIES(${LIBRARY_OUTPUT_DIRECTORY})
+add_library(avtdatabase_ser)
```
this looks nice! I would add a comment to these libs that states the sources are added via cmake logic that `add_subdirectory` pulls in, just to make it clear here where the heavy lifting is happening.

biagas replied:
I could add a section called 'CMake conventions' to our style guide, would that suffice?

cyrush replied:
Yes, that sounds like a better place to describe the strategy.

biagas replied:
I added ticket #18426 to keep track of documenting.

cyrush reviewed (dismissed):
Looks good. One suggestion about adding comments about the nature of add_lib + add_subdir. Comment might help folks understand how to interpret the logic, but this is a repeated pattern -- maybe it belongs in docs instead of comments at every instance?

biagas committed:
Merge branch 'develop' into task/biags/cmake_reorg_cmakelists_each_dir

JustinPrivitera reviewed (dismissed):
I don't know if I'm qualified at this point to approve a cmake-related PR, but it looks good as far as I can tell.

biagas committed:
Add CMake conventions section to StyleGuide.

Preliminary effort, explains the logic where a target's sources are spread across multiple subdirectories.

biagas committed:
Merge branch 'develop' into task/biags/cmake_reorg_cmakelists_each_dir

biagas commented:
@cyrush  I added a new 'CMake conventions' to the Style Guide doc. Just explains the subirectory logic right now. What do you think?

cyrush commented on src/doc/dev_manual/StyleGuide.rst:
```diff
@@ -641,3 +641,26 @@ Example: ::
+The subdirectory will add its sources to the parent's target via `target_sources`.
+
+
+Here's an example from src/avt/DBAtts and src/avt/DBAtts/SIL:
```
Looks good!

biagas committed:
Merge branch 'develop' into task/biags/cmake_reorg_cmakelists_each_dir
